tests/pytest/snapshot_test.py

```diff
@@ -0,0 +1,25 @@
+from test_dragonfly import DRAGONFLY_PATH, dfly_args, df_server, client, connection
+
+import redis
+from pathlib import Path
+import time
+import pytest
+
+OUT_DIR = Path(DRAGONFLY_PATH).parent
+
+@pytest.mark.usefixtures("client")
+@dfly_args("--dbfilename", "test.rdb", 
+	"--save_schedule", "*:*", 
+	"--dir", str(OUT_DIR)+"/")
+class TestSnapshot:
+	def test_snapshot(self, client: redis.Redis):
+		out_path = OUT_DIR / "test.rdb"
+		client.set("test", "test")
+
+		if out_path.exists():
```

Comment:
Lets make it properly (you are the bar raiser, and others will just copy your code 😺 ):

run cleanup before the test:
https://stackoverflow.com/questions/22627659/run-code-before-and-after-each-test-in-py-test

make the fixture class-level:
https://stackoverflow.com/questions/50132703/pytest-fixture-for-a-class-through-self-not-as-method-argument

Reply:
One of the issues I ran into was that pytest appears to have a long standing limitation in passing fixtures to `pytest.mark.parametrize` function(https://github.com/pytest-dev/pytest/issues/349). Since the dragonfly args functions are essentially wrappers over this, making a temporary directory fixture required a bit of a hacky workaround to be used both in the subprocess arguments and as a fixture in tests.

What I ended up deciding on is to run the dragonfly subprocess in a clean temporary directory (which is then cleaned up at the end of the session). Arguments can access the path of the temporary directory by using standard python formatting and tests can access the temporary directory fixture as any other.

Reply:
Sounds good. It's beyond my pytest knowledge, but the code now looks order of magnitude better than before! :)